hermes/probe/mock/mock_gcs.go

```diff
@@ -0,0 +1,88 @@
+package mock
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"cloud.google.com/go/storage"
+	"github.com/googleapis/google-cloud-go-testing/storage/stiface"
+)
+
+type fakeBucket struct {
+	attrs   *storage.BucketAttrs
+	objects map[string][]byte
+}
+
+type fakeClient struct {
+	stiface.Client
+	buckets map[string]*fakeBucket
+}
+
+type fakeBucketHandle struct {
+	stiface.BucketHandle
+	c    *fakeClient
+	name string
+}
+
+type fakeObjectHandle struct {
+	stiface.ObjectHandle
+	c          *fakeClient
+	bucketName string
+	name       string
+}
+
+type fakeWriter struct {
+	stiface.Writer
+	obj fakeObjectHandle
+	buf bytes.Buffer
+}
+
+func NewFakeClient() stiface.Client {
```

Comment:
Return values should be concrete types instead of interfaces (interfaces are to be used as the type of arguments).
Make `fakeClient` exported and change the return type of this function to `*FakeClient`.

Reply:
But then if NewFakeClient() returns *FakeClient and in arguments for create file (which we are among others testing using this)  we have stiface.Client how can we pass *FakeClient as stiface.Client?

Reply:
If `*FakeClient` satisfies the `stiface.Client` interface, then it will be accepted as such. This is the point of interfaces in Go.

Reply:
It does satifsy the stiface.Client and it works. Thank you.